src/execution_plan/optimizations/traverse_order.c

```diff
@@ -107,23 +107,23 @@ static bool _valid_arrangement(const Arrangement arrangement, uint exps_count) {
 	 * exp2: [L1]
 	 * Isn't valid, as currently the first expression is converted
 	 * into a scan operation. */
-	if((exp->src_node->label || exp->dest_node->label) &&
+	QGNode *src = QueryGraph_GetNodeByAlias(qg, exp->src); // TODO unwisely expensive
+	QGNode *dest = QueryGraph_GetNodeByAlias(qg, exp->dest); // TODO unwisely expensive
```

Comment:
The QueryGraph lookups in this function (in order to check if src and dest nodes are labeled) are not great. They should definitely not be invoked in the more expensive loops, though we could possibly make this improvement when replacing the factorial logic here instead of now. Thoughts?

Reply:
If we find this to be indeed wasteful, we might improve by introducing a `rax` for quick querygraph entity lookup.